modules/snmp/snmp.go

```diff
@@ -135,13 +135,21 @@ func (s *SNMP) Init() bool {
 	s.SNMPClient = snmpClient
 
 	if len(s.ChartInput) > 0 {
-		s.charts = newChart(s.ChartInput)
+		s.charts, err = allCharts(s.ChartInput)
+		if err != nil {
+			s.Errorf("Population of charts failed: %v", err)
+			return false
+		}
 	} else {
 		c := defaultSNMPchart.Copy()
 		c.ID = fmt.Sprintf(c.ID, 1)
 		c.Title = fmt.Sprint(c.Title, "default")
-		_ = c.AddDim(defaultDims[0])
-		_ = c.AddDim(defaultDims[1])
+		for _, d := range defaultDims {
+			if err = c.AddDim(d); err != nil {
+				s.Errorf("Population of charts failed: %v", err)
+				return false
+			}
+		}
 		s.charts = &module.Charts{c}
 	}
```

Comment:
I don't think we need that `else` case with the default chart. The assumption that we need to collect ifInOctets/ifOutOctets for interface with index 2 (why 2?) looks pretty questionable to me.

Reply:
I see that it is not a new code. So we can merge it as is, but let's remove this case right away in the following PR.

Reply:
Perhaps I am missing something, if so please elaborate on why you think we need that.

Reply:
Ok, added it to https://github.com/netdata/go.d.plugin/issues/648#issuecomment-1057198748